src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -50,6 +53,62 @@ internal class PackageLoadContext
 
         public Task<PackageCollection> GetInstalledPackagesAsync() =>_installedPackagesTask;
 
+        // get lock file for projects
+        public async Task<List<LockFile>> getLockFiles(IEnumerable<NuGetProject> nuGetProjects)
+        {
+            var nuGetProjectList = nuGetProjects.ToList();
+            var buildIntegratedProjects = nuGetProjectList.OfType<BuildIntegratedNuGetProject>().ToList();
+
+            List<LockFile> lockFiles = new List<LockFile>();
+            var lockFileFormat = new LockFileFormat();
+
+            foreach (var project in buildIntegratedProjects)
+            {
+                var lockFilePath = await project.GetAssetsFilePathAsync();
+                if (File.Exists(lockFilePath))
+                {
+                    lockFiles = lockFiles.Append(lockFileFormat.Read(lockFilePath)).ToList();
```

Comment:
This incurs a very expensive performance hit. 
I'm not sure how I'd want us to do this yet, but we should consider caching it, at least for the context of the PM UI.

Reply:
Yes, transitive dependencies are used by the recommender. We don't want to recommend a package that is already included as a dependency.